taipy/gui/data/decimator/base.py

```diff
@@ -0,0 +1,282 @@
+# Copyright 2021-2024 Avaiga Private Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+# the License. You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+# an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+from __future__ import annotations
+
+import typing as t
+from abc import ABC, abstractmethod
+
+import numpy as np
+import pandas as pd
+
+from ..._warnings import _warn
+
+
+class Decimator(ABC):
+    """Base class for decimating chart data.
+
+    *Decimating* is the term used to name the process of reducing the number of
+    data points displayed in charts while retaining the overall shape of the traces.
+    `Decimator` is a base class that does decimation on data sets.
+
+    Taipy GUI comes out-of-the-box with several implementation of this class for
+    different use cases.
+    """
+
+    _CHART_MODES: t.List[str] = []
+
+    def __init__(
+        self,
+        threshold: t.Optional[int],
+        zoom: t.Optional[bool],
+        on_decimate: t.Optional[t.Callable] = None,
+        apply_decimator: t.Optional[t.Callable] = None,
+    ) -> None:  # noqa: E501
+        """Initialize a new `Decimator`.
+
+        Arguments:
+            threshold (Optional[int]): The minimum amount of data points before the
+                decimator class is applied.
+            zoom (Optional[bool]): set to True to reapply the decimation
+                when zoom or re-layout events are triggered.
+            on_decimate (Optional[Callable]): an user-defined function that is executed when the decimator
```

Comment:
Please remove the doc for these parameters - place them in a comment elsewhere so the doc doesn't mess up.